### PR TITLE
Use `monkeypatch.chdir` instead of `os.chdir` in tests

### LINKED ISSUE #15576: Use `monkeypatch.chdir` instead of `os.chdir` in tests
### Motivation

Many of the tests under tests/tests_app use the following pattern:

```py
cwd = os.getcwd()
os.chdir(new_cwd)

# some test code

os.chdir(cwd)
```

This is not ideal. If any one of these tests fails, the current working directory remains to what it was set in the beginning of the code. This can cause subsequent tests to fail for unrelated reasons.

### Pitch

Use 

```py
monkeypatch.chdir(new_cwd)
```

instead.

cc @tchaton 

cc @borda @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by heytanay**
## What does this PR do?
This PR changes `os.chdir()` calls to `monkeypatch.chdir()` in test as requested in issue #15576 

Fixes #15576 

### Does your PR introduce any breaking changes? If yes, please list them.
No

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review
@Borda @awaelchli 

## Did you have fun?
Yes!

### CODE REVIEW

File: tests/tests_app/cli/test_cmd_install.py [hunk 1]
```diff
@@ -61,11 +61,11 @@ def test_valid_unpublished_app_name():
 
 
 @pytest.mark.skip(reason="need to figure out how to authorize git clone from the private repo")
-def test_app_install(tmpdir):
+def test_app_install(tmpdir, monkeypatch):
     """Tests unpublished app install."""
 
     cwd = os.getcwd()
-    os.chdir(tmpdir)
+    monkeypatch.chdir(tmpdir)
```
Comment (awaelchli) [inline tests/tests_app/cli/test_cmd_install.py:68]: we should also remove the `os.chdir(...)` at the end of the test (in all the tests that have it). When using the monkeypatch, this teardown logic is no longer required.

Comment (heytanay) [reply tests/tests_app/cli/test_cmd_install.py:68]: Right! Sorry I missed it.

Comment (heytanay) [reply tests/tests_app/cli/test_cmd_install.py:68]: @awaelchli Thanks, I have removed it.

Comment (heytanay) [reply tests/tests_app/cli/test_cmd_install.py:67]: Hi, are there any changes to make that I might have missed?

Comment (awaelchli) [reply tests/tests_app/cli/test_cmd_install.py:67]: I added a few more things that was missed. No big deal. Let's see if tests pass and then we should be good! Thanks

Comment (heytanay) [reply tests/tests_app/cli/test_cmd_install.py:67]: Thanks for the changes! It's my first contribution to Lightning so I was a bit confused throughout 😅

**Review by awaelchli — APPROVED** [review]

File: tests/tests_app/cli/test_cmd_install.py (hunk 1 above)
Comment (awaelchli) [reply tests/tests_app/cli/test_cmd_install.py:67]: Perfectly normal 🎉  and we're here to help :)

**Review by tchaton — APPROVED** [review]
Looks great ! Thanks.

**Pull request merged**